packages/js/src/dashboard/widgets/search-ranking-compare-widget.js

```diff
@@ -0,0 +1,118 @@
+import { useCallback, useMemo } from "@wordpress/element";
+import { useRemoteData } from "../services/use-remote-data";
+import { Widget } from "./widget";
+import { getDifference } from "../transformers/difference";
+import { SearchRankingCompareWidgetContent } from "./search-ranking-compare/search-ranking-compare-widget-content";
+/**
+ * @typedef {Object<TimeFrame, TimeFrameData>} TimeBasedData The time based data.
+ */
+
+/**
+ * @typedef { "impressions"|"clicks"|"ctr"|"position" } SearchRankingCompareMetric The widget type.
+ */
+
+/**
+ * @typedef {Object} MetricData
+ * @property {number} value - The value of the metric.
+ * @property {number} delta - The delta of the metric.
+ */
+
+/**
+ * @typedef {Object<SearchRankingCompareMetric, MetricData | null>} SearchRankingCompareData The search ranking compare data.
+ * @property {MetricData} impressions - The impressions data.
+ * @property {MetricData} clicks - The clicks data.
+ * @property {MetricData} ctr - The click-through rate data (optional).
+ * @property {MetricData} position - The average position data (optional).
+ */
+
+/* eslint-disable complexity */
+/**
+ * @param {TimeBasedData[]} data The data.
+ * @returns {SearchRankingCompareData|TimeBasedData[]} The transformed data.
+ */
+const transformData = ( data ) => {
+	if ( data.length === 0 ) {
+		return data;
+	}
+	if ( Object.keys( data[ 0 ].current ).length === 2 ) {
+		return {
+			impressions: {
+				value: data[ 0 ].current.total_impressions || NaN,
+				delta: getDifference( data[ 0 ]?.current?.total_impressions || NaN, data[ 0 ]?.previous?.total_impressions || NaN ),
+			},
+			clicks: {
+				value: data[ 0 ].current.total_clicks || NaN,
+				delta: getDifference( data[ 0 ]?.current?.total_clicks || NaN, data[ 0 ].previous.total_clicks || NaN ),
+			},
+		};
+	}
+
+	return {
+		impressions: {
+			value: data[ 0 ].current.total_impressions || NaN,
+			delta: getDifference( data[ 0 ].current.total_impressions || NaN, data[ 0 ].previous.total_impressions || NaN ),
+		},
+		clicks: {
+			value: data[ 0 ].current.total_clicks || NaN,
+			delta: getDifference( data[ 0 ].current.total_clicks || NaN, data[ 0 ].previous.total_clicks || NaN ),
+		},
+		ctr: {
+			value: data[ 0 ].current.average_ctr || NaN,
+			delta: getDifference( data[ 0 ].current.average_ctr || NaN, data[ 0 ].previous.average_ctr || NaN ),
+		},
+		position: {
+			value: data[ 0 ].current.average_position || NaN,
+			delta: getDifference( data[ 0 ].current.average_position || NaN, data[ 0 ].previous.average_position || NaN ),
+		},
+	};
+};
+/* eslint-enable complexity */
+
+/**
+ * @param {import("../services/comparison-metrics-data-formatter")} dataFormatter The data formatter.
+ * @returns {function(?TimeBasedData[]): SearchRankingCompareData} Function to format the widget data.
```

Comment:
This return type is actually not correct. The formatter changes it from `MetricData` to some other structure you have not defined, with `formattedValue` etc.

Reply:
Added the `FormattedMetricData` type.